src/main/kotlin/net/maizegenetics/phgv2/cli/Hvcf2Gvcf.kt

```diff
@@ -0,0 +1,571 @@
+    private val myLogger = LogManager.getLogger(Hvcf2Gvcf::class.java)
+
+    val hvcfDir by option("--hvcf-dir", help = "Path to directory holding hVCF files. Data will be pulled directly from these files instead of querying TileDB")
+        .default("")
+        .validate {
+            require(it.isNotBlank()) {
+                "--hvcf-dir must not be blank"
+            }
+        }
+
+    val condaEnvPrefix by option (help = "Prefix for the conda environment to use.  If provided, this should be the full path to the conda environment.")
+        .default("")
+
+    val outputDir by option (help = "Output directory for the gVCF files.  If not provided, the current working directory is used.")
+        .default("")
+
+    val dbPath by option(help = "Folder name where TileDB datasets and AGC record is stored.  If not provided, the current working directory is used")
+        .default("")
+    val referenceFile by option(help = "Path to local Reference FASTA file needed for sequence dictionary")
+        .default("")
+        .validate {
+            require(it.isNotBlank()) {
+                "--reference-file must not be blank"
+            }
+        }
+
+    override fun run() {
+        val dbPath = if (dbPath.isBlank()) {
+            System.getProperty("user.dir")
+        } else {
+            dbPath
+        }
+
+        myLogger.info("Hvcf2Gvcf: dbPath = $dbPath, hvcfDir = $hvcfDir, outputDir = $outputDir, condaEnvPrefix = $condaEnvPrefix")
+
+        // Verify the tiledbURI - verifyURI will throw an exception if the URI is not valid
+        val validDB = verifyURI(dbPath,"hvcf_dataset",condaEnvPrefix)
+
+        // read and store ref file data for later use when creating the vcf sequence dictionary
+        val refSeq = CreateMafVcf().buildRefGenomeSeq(referenceFile)
+        buildGvcfFromHvcf(dbPath, refSeq, outputDir, hvcfDir, condaEnvPrefix)
+    }
+
+    fun buildGvcfFromHvcf(dbPath: String, refSeq: Map<String, NucSeq>, outputDir: String, hvcfDir: String, condaEnvPrefix: String) {
+        // get list of hvcf files
+        // walk the gvcf directory process files with g.vcf.gz extension
+        File(hvcfDir).walk().filter { !it.isHidden && !it.isDirectory }
+            .filter { it.name.endsWith("h.vcf.gz")  || it.name.endsWith("h.vcf")  }.toList()
+            .forEach { hvcfFile ->
+                myLogger.info("buildGvcfFromHvcf: Processing hvcf file: ${hvcfFile.name}")
+                val records = processSingleHVCF(refSeq, outputDir, hvcfFile, dbPath,condaEnvPrefix)
+
+                val sample = hvcfFile.toString().substringAfterLast("/").substringBefore(".")
+                val gvcfFile = "$outputDir/${sample}.g.vcf"
+                exportVariantContext(sample,records,gvcfFile, refSeq,setOf())
+            }
+
+    }
+
+    fun processSingleHVCF(refSeq:Map<String,NucSeq>, outputDir:String, hvcfFile: File, dbPath: String, condaEnvPrefix: String): List<VariantContext> {
```

Comment:
Can this have a more specific name?  Maybe processSingleHVCFIntoVariantContext or something a bit simpler?

Reply:
I changed to processHVCFtoVariantContext